bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FtpFolderWatcherHandler.java

```diff
@@ -0,0 +1,263 @@
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPFile;
+import org.apache.commons.net.ftp.FTPReply;
+import org.apache.commons.net.ftp.FTPSClient;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.folderwatcher.internal.common.WatcherCommon;
+import org.openhab.binding.folderwatcher.internal.config.FtpFolderWatcherConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FtpFolderWatcherHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexandr Salamatov - Initial contribution
+ */
+@NonNullByDefault
+public class FtpFolderWatcherHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(FtpFolderWatcherHandler.class);
+    private FtpFolderWatcherConfiguration config = new FtpFolderWatcherConfiguration();
+    private @Nullable File currentFtpListingFile;
+    private @Nullable ScheduledFuture<?> executionJob, initJob;
+    private FTPClient ftp = new FTPClient();
+    private List<String> previousFtpListing = new ArrayList<>();
+
+    public FtpFolderWatcherHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Channel {} triggered with command {}", channelUID.getId(), command);
+        if (command instanceof RefreshType) {
+            refreshFTPFolderInformation();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        File currentFtpListingFile;
+        config = getConfigAs(FtpFolderWatcherConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        if (config.connectionTimeout <= 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Connection timeout can't be negative");
+            return;
+        }
+        if (config.ftpPort < 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "FTP port can't be negative");
+            return;
+        }
+        if (config.pollInterval <= 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Polling interval can't be null or negative");
+        }
+
+        if (!config.secureMode.matches("(?i)NONE|IMPLICIT|EXPLICIT")) {
```

Comment:
I think it would be better to create an enum for that. You can look https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/config/BaseConfig.java and https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/ServerSecurity.java how to do that

Reply:
@J-N-K great, in that case I won't need this check at all